### PR TITLE
Added tests to check for aria labels and roles for accessibility

### PR DESCRIPTION
**Pull request by m158261**
## References

Related PR - https://github.com/jupyterlab/jupyterlab/pull/9622

## Code changes

Added tests to check aria labels and roles are present for users with screen readers. There a mostly around sidebar widgets. Some of the tests are run in Galata due to restrictions with rendering extension packages. 

## User-facing changes

None

## Backwards-incompatible changes

None

### CODE REVIEW

File: packages/extensionmanager-extension/src/index.ts
Comment (krassowski) [inline packages/extensionmanager-extension/src/index.ts:60]: What is `data-testid`?

Comment (m158261) [reply packages/extensionmanager-extension/src/index.ts:60]: It's used by getByTestId in Galata to get the extension manager section. I tried using page.locator() with the id 'extensionmanager.main-view' but that returned multiple elements.

Comment (krassowski) [reply packages/extensionmanager-extension/src/index.ts:60]: In the commit history I see that you tried `getByTitle` but I do not see you using the ID locator directly (instead of using `getByTitle`. I think this is not needed - can you give using just ID another try please?

Comment (m158261) [reply packages/extensionmanager-extension/src/index.ts]: Hey @krassowski, I've tried just using the ID locator and that can't find the element. If I try and click on the extension manager it returns two elements. I've tried first() and last() but those time out too. I think it's because there are two elements that have an id or data-id of extensionmanager.main-view. Adding the data-testid was the solution that I found that worked.

Comment (krassowski) [reply packages/extensionmanager-extension/src/index.ts]: Do you see two elements with `extensionmanager.main-view` ID in the page? That would be a pretty serious bug. My guess is that there was some mistake in the selector, for example unescaped dot `.`. Did you try `#extensionmanager\\.main-view`?

Comment (m158261) [reply packages/extensionmanager-extension/src/index.ts]: The extension manager section has an ID of extensionmanager.main-view. The extension manager li has a data-id of extensionmanager.main-view. 

I'll try it with an escaped character. Thanks, I didn't think of that.

File: packages/application/test/shell.spec.ts
Comment (krassowski) [inline packages/application/test/shell.spec.ts:521]: I would name the group `accessibility` in case if we want to add more tests which do not fit neatly to specific methods or attributes in the future.

File: packages/apputils/test/mainareawidget.spec.ts
Comment (krassowski) [inline packages/apputils/test/mainareawidget.spec.ts:41]: This test demonstrated incorrect behaviour (not every main area widget is a notebook, for example a text editor is not a notebook, csv viewer is not a notebook etc), see https://github.com/jupyterlab/jupyterlab/issues/13045. I am not sure what the default for a main area widget should be but my expectation is that it would not be "notebook". Also, is the toolbar really `navigation`? Shouldn't it be a [toolbar role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/toolbar_role)?

Comment (m158261) [reply packages/apputils/test/mainareawidget.spec.ts:41]: I can change the navigation role to toolbar. As the notebook content label still has an outstanding issue, I could either remove the test or leave it to be updated when that issue is worked on?

File: packages/application/test/shell.spec.ts
Comment (m158261) [reply packages/application/test/shell.spec.ts]: Changes made.

File: packages/extensionmanager-extension/src/index.ts
Comment (m158261) [reply packages/extensionmanager-extension/src/index.ts]: The test now passes after adding an escaped character.

File: packages/apputils/test/mainareawidget.spec.ts
Comment (krassowski) [reply packages/apputils/test/mainareawidget.spec.ts]: Let's keep the test, but:
- mark it with [`test.failing`](https://jestjs.io/docs/api#testfailingname-fn-timeout) as xfail for now
- add a comment on the test pointing to issue
- once this PR is merged, comment on the issue that there is now a test.

Comment (m158261) [reply packages/apputils/test/mainareawidget.spec.ts]: Changes made.

**Review by krassowski — APPROVED** [review]
Thank you  @m158261!

**Pull request merged**
